From 784277e395c7db1e3d239a4661ab76efc4c34060 Mon Sep 17 00:00:00 2001
From: Michael Auchter <michael.auchter@ni.com>
Date: Tue, 18 Jun 2019 11:54:14 -0500
Subject: [PATCH 02/11] Revert "mmc: Downgrade SD/MMC from UHS/HS200/HS400
 modes before boot"

This reverts commit fceea9926804c8e72bdcde27bf64bebedcf0a754.

On the X410, we want to use HS200 mode for the eMMC. This commit added
code that would change the eMMC mode out of HS200 prior to booting. For
ZynqMP, this causes the zynq_sdhci driver to program different tap
delays for the slower mode.

This is problematic because Linux does not program these tap delays, and
yet expects to use the eMMC in HS200 mode. This results in very unstable
access to the eMMC in Linux.

Revert this commit to leave the eMMC in HS200 mode (and with the
appropriate tap delays).

Signed-off-by: Michael Auchter <michael.auchter@ni.com>
---
 drivers/mmc/mmc-uclass.c | 19 -------------------
 drivers/mmc/mmc.c        | 26 --------------------------
 include/mmc.h            |  6 ------
 3 files changed, 51 deletions(-)

diff --git a/drivers/mmc/mmc-uclass.c b/drivers/mmc/mmc-uclass.c
index 90690c8d1e..484bcaeed5 100644
--- a/drivers/mmc/mmc-uclass.c
+++ b/drivers/mmc/mmc-uclass.c
@@ -437,19 +437,6 @@ static int mmc_blk_probe(struct udevice *dev)
 	return 0;
 }
 
-#if CONFIG_IS_ENABLED(MMC_UHS_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS200_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS400_SUPPORT)
-static int mmc_blk_remove(struct udevice *dev)
-{
-	struct udevice *mmc_dev = dev_get_parent(dev);
-	struct mmc_uclass_priv *upriv = dev_get_uclass_priv(mmc_dev);
-	struct mmc *mmc = upriv->mmc;
-
-	return mmc_deinit(mmc);
-}
-#endif
-
 static const struct blk_ops mmc_blk_ops = {
 	.read	= mmc_bread,
 #if CONFIG_IS_ENABLED(MMC_WRITE)
@@ -464,12 +451,6 @@ U_BOOT_DRIVER(mmc_blk) = {
 	.id		= UCLASS_BLK,
 	.ops		= &mmc_blk_ops,
 	.probe		= mmc_blk_probe,
-#if CONFIG_IS_ENABLED(MMC_UHS_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS200_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS400_SUPPORT)
-	.remove		= mmc_blk_remove,
-	.flags		= DM_FLAG_OS_PREPARE,
-#endif
 };
 #endif /* CONFIG_BLK */
 
diff --git a/drivers/mmc/mmc.c b/drivers/mmc/mmc.c
index d79cdef62e..00cf3a2664 100644
--- a/drivers/mmc/mmc.c
+++ b/drivers/mmc/mmc.c
@@ -2944,32 +2944,6 @@ int mmc_init(struct mmc *mmc)
 	return err;
 }
 
-#if CONFIG_IS_ENABLED(MMC_UHS_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS200_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS400_SUPPORT)
-int mmc_deinit(struct mmc *mmc)
-{
-	u32 caps_filtered;
-
-	if (!mmc->has_init)
-		return 0;
-
-	if (IS_SD(mmc)) {
-		caps_filtered = mmc->card_caps &
-			~(MMC_CAP(UHS_SDR12) | MMC_CAP(UHS_SDR25) |
-			  MMC_CAP(UHS_SDR50) | MMC_CAP(UHS_DDR50) |
-			  MMC_CAP(UHS_SDR104));
-
-		return sd_select_mode_and_width(mmc, caps_filtered);
-	} else {
-		caps_filtered = mmc->card_caps &
-			~(MMC_CAP(MMC_HS_200) | MMC_CAP(MMC_HS_400));
-
-		return mmc_select_mode_and_width(mmc, caps_filtered);
-	}
-}
-#endif
-
 int mmc_set_dsr(struct mmc *mmc, u16 val)
 {
 	mmc->dsr = val;
diff --git a/include/mmc.h b/include/mmc.h
index 82562193cc..403ab3b46d 100644
--- a/include/mmc.h
+++ b/include/mmc.h
@@ -749,12 +749,6 @@ int mmc_init_device(int num);
 int mmc_init(struct mmc *mmc);
 int mmc_send_tuning(struct mmc *mmc, u32 opcode, int *cmd_error);
 
-#if CONFIG_IS_ENABLED(MMC_UHS_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS200_SUPPORT) || \
-    CONFIG_IS_ENABLED(MMC_HS400_SUPPORT)
-int mmc_deinit(struct mmc *mmc);
-#endif
-
 /**
  * mmc_of_parse() - Parse the device tree to get the capabilities of the host
  *
-- 
2.29.2

